controllers/datamover.go

```diff
@@ -186,3 +206,158 @@ func (r *DPAReconciler) getDataMoverLabels() map[string]string {
 	labels[oadpv1alpha1.DataMoverDeploymentLabel] = "True"
 	return labels
 }
+
+// Check if there is a valid user supplied restic secret
+func (r *DPAReconciler) validateDataMoverCredential(resticsecret *corev1.Secret) bool {
+	if resticsecret == nil {
+		return false
+	}
+	for key, val := range resticsecret.Data {
+
+		if key == ResticPassword {
+			if len(val) == 0 {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func (r *DPAReconciler) createResticSecretsPerBSL(dpa *oadpv1alpha1.DataProtectionApplication, bsl velerov1.BackupStorageLocation, dmresticsecretname string, pass []byte) (*corev1.Secret, error) {
+
+	switch bsl.Spec.Provider {
+	case AWSProvider:
+		{
+			secretName, secretKey := r.getSecretNameAndKey(&bsl.Spec, oadpv1alpha1.DefaultPluginAWS)
+			bslSecret, err := r.getProviderSecret(secretName)
+			if err != nil {
+				return nil, err
+			}
+
+			awsProfile := "default"
+
+			if value, exists := bsl.Spec.Config[Profile]; exists {
+				awsProfile = value
+			}
+
+			key, secret, err := r.parseAWSSecret(bslSecret, secretKey, awsProfile)
+			if err != nil {
+				r.Log.Info(fmt.Sprintf("Error parsing provider secret %s for backupstoragelocation", secretName))
+				return nil, err
+			}
+			repobase := "s3:s3.amazonaws.com/"
+			if bsl.Spec.Config["s3Url"] != "" {
+				repobase = bsl.Spec.Config["s3Url"]
+			}
+
+			repo := repobase + bsl.Spec.ObjectStorage.Bucket
+			rsecret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("%s-volsync-restic", bsl.Name),
+					Namespace: bsl.Namespace,
+					Labels: map[string]string{
+						oadpv1alpha1.OadpBSLnameLabel: bsl.Name,
```

Comment:
As the secret is create by OADP Operator, maybe add the operatort label as well, so that we do not miss on the reconciles via events on these secrets